FEAT: add method to move topics between streams

tmichela opened the pull request:
Move a topic from one stream to another,

Example usage:
```python
>>> client.move_topic('stream_a', 'stream_b', 'my_topic')
{'result': 'success', 'msg': ''}
```

timabbott commented on zulip/zulip/__init__.py:
```diff
@@ -1554,6 +1554,53 @@ def set_typing_status(self, request: Dict[str, Any]) -> Dict[str, Any]:
+        Example usage::
+
+            >>> client.move_topic('stream_a', 'stream_b', 'my_topic')
+            {'result': 'success', 'msg': ''}
```
This is over-indented, I think.

tmichela replied:
I've changed the docstring to match the exiting style

timabbott commented on zulip/zulip/__init__.py:
```diff
@@ -1554,6 +1554,53 @@ def set_typing_status(self, request: Dict[str, Any]) -> Dict[str, Any]:
+        stream: str,
+        new_stream: str,
+        topic: str,
+        new_topic: Optional[str] = None
```
Ideally, this would support the `propagate_mode` option and a message ID threshold.  https://zulipchat.com/api/update-message#parameters has some basic documentation

timabbott commented on zulip/zulip/__init__.py:
```diff
@@ -1554,6 +1554,53 @@ def set_typing_status(self, request: Dict[str, Any]) -> Dict[str, Any]:
+            'num_before': 1,
+            'num_after': 0,
+        }
+        reply = self.get_messages(request)
```
Let's call this `result`; we use that local variable name for API results in the codebase.

Also, can you add a few blank lines in this function to match the style of the codebase?

timabbott commented:
@tmichela thanks for working on this!  I posted a few comments.  It'd also be great if you wrote a proper commit message; 
Check out the Zulip commit message guidelines for more details: https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages.

@akashaviator can you do a review pass on this as well?

tmichela commented:
@timabbott thanks for the review and suggestions!

I think I've addressed your comments.

I also changed/fixed:
- get stream IDs from their name (got some issues while testing using the stream names, returning sometimes an error)
- added arguments for `message_id` and `propagate_mode` as you suggested
- added arguments to silence notification to old/new topic

timabbott commented:
This is great, merged, thanks @tmichela!  

Would you be up for submitting a PR to document this in the examples in https://zulip.com/api/update-message?  I expect https://zulip.readthedocs.io/en/latest/documentation/api.html should cover the background you need; let me know if it's missing anything!

tmichela commented:
Sure, I'll look into this.

Do you want me to simply update the python example code?
It looks like this documentation reflects more of the direct restAPI call while this method wraps a few different calls to make it more convenient to use.

timabbott commented:
Yeah, exactly; I'd just include it in the same block where we document the more messy/direct way to do it.
